Automattic/jetpack
# [WooCommerce Analytics] Support for Search and Landing Page

## Proposed changes:

* Add support for `woocommerceanalytics_search` event 
* Rename `woocommerceanalytics_session_id` into `woocommerceanalytics_session` since now it stores more data
* Refactor session data adding landing_page to it.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






Prerequisites

* Download and load Tracks Vigilante Chrome extension https://github.com/Automattic/tracks-chrome-extension
* Install Woo and Jetpack with this PR
* Connect Jetpack and WP.com

--

1. Enter any page on the site and verify `woocommerceanalytics_session_started` is triggered with a `session_id` and `landing_page` prop with a clean URL.
2. Navigate further over a product detail page and verify the events are trigger with the same `session_id` and `landing_page` props
3. Perform a search (can be done by URL using `?s={query}` or by adding Search Products block somewhere)
4. See  `woocommerceanalytics_search` is triggered with a `search_query` prop indicating the search query and `qty` prop indicating the results count. Also, `session_id` and `landing_page` props match the expected.
5. Change your permalinks structure to use a Plain structure 
6. Remove `woocommerceanalytics_session` session cookie
7. Repeat step 1. This structure should work as well.
